src/autoscrapper/interaction/ui_windows.py

```
@@ -85,7 +81,16 @@ def wait_for_target_window(
         win = pwc.getActiveWindow()
         if win is not None:
             app = (win.getAppName() or "").lower()
-            if app == target_lower:
+            title = ""
+            if hasattr(win, "title"):
+                title = getattr(win, "title") or ""
```

Ruff (0.14.10)

Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)